mobify/progressive-web-scaffold
# Open Sans: remove; Oswald: switch from 300 → 200

Reduce number of Google fonts being loaded.

 **JIRA**: [WEB-1248](https://mobify.atlassian.net/browse/WEB-1248)

## Changes
- Replace Open Sans with the following system fonts: `'San Francisco', 'Roboto', 'Fira Sans', 'Segoe UI'`
- Oswald: remove weight 700, and replace 300 (light) with 200 (extra light)
- Replace all current usages of `u-text-lighter` with new `u-text-extra-lighter` class

| Before | After |
| ------- | ----- |
| ![screen shot 2017-03-10 at 4 28 06 pm](https://cloud.githubusercontent.com/assets/734535/23818339/ef088298-05ae-11e7-9a8d-a2fb1ff42324.png) | ![screen shot 2017-03-10 at 4 22 54 pm](https://cloud.githubusercontent.com/assets/734535/23818342/f3d5fcb0-05ae-11e7-976e-c8d06ddbd030.png) |

## Impact
The above screenshots don't suggest that much, if any, impact happens by this change; just that one less font file is downloaded.

The more important change will come when we fix how we avoid FOIT in [a near future update](https://mobify.atlassian.net/browse/WEB-1249), because as can be seen in the below screenshot, slow loaded fonts clearly prevent the site from showing meaningful content until after they finish loading.

> Slow fonts = slow meaningful first paint

| Slow Loaded Fonts (block rendering) | Fast Loaded Fonts (minimal blocking) |
| ------------------------------------- | ------------------------------------- |
| ![screen shot 2017-03-13 at 2 00 47 pm](https://cloud.githubusercontent.com/assets/734535/23875805/fc904964-07f7-11e7-81f5-86030bd46481.png) | ![screen shot 2017-03-13 at 2 12 55 pm](https://cloud.githubusercontent.com/assets/734535/23875811/018f4654-07f8-11e7-815b-525d1299785c.png) |

Additionally, it's the call to Google's font CSS file (that contains the `@font-faces` for our fonts) that potentially slow us down, or at least consume more bandwidth than is necessary. We may want to explore cutting out the 2-3 seconds spent downloading assets from `fonts.googleapis.com` if possible, or defer to a latter time.

![screen shot 2017-03-10 at 4 45 31 pm](https://cloud.githubusercontent.com/assets/734535/23818959/44e27d12-05b5-11e7-9e3f-0a0581b7ea4b.png)

What if we just **removed fonts all together?** Still not a drastic change, as you can see in the below screenshot.

![screen shot 2017-03-13 at 5 14 25 pm](https://cloud.githubusercontent.com/assets/734535/23880424/d7a90b04-0810-11e7-8f0d-745ee7f96a7a.png)
> __(Note: Web Page Test of a preview bundle, so not truly indicative of prod performance)__

### Takeaways

- [ ] Defer loading fonts to later
- [ ] Avoid requesting `fonts.googleapis.com` CSS files if possible
- [ ] [Avoid `@font-face`](https://www.zachleat.com/web/comprehensive-webfonts/#font-face)
- [ ] Use the [FOUT w/ Class technique](https://www.zachleat.com/web/comprehensive-webfonts/#fout-class) or [FOFT w/ 2 Stage Render](https://www.zachleat.com/web/comprehensive-webfonts/#foft)

Some of the above may be addressable with [Web Font Loader](https://github.com/typekit/webfontloader)

That said, the benefits of these changes may be minimal:

- Perceived completion by the presents of text may only boost by ~2-3%
- Rendered text may only appear ~0.2-0.6 seconds earlier

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Verify that heading fonts look correct across the site
